src/inmanta/server/services/orchestrationservice.py

```diff
@@ -165,18 +174,22 @@ def pair_resources_partial_update_to_old_version(
             paired_resources.append(pair)
         return paired_resources
 
-    async def _get_old_resources(
-        self,
-    ) -> Dict[ResourceIdStr, ResourceWithResourceSet]:
+    async def _get_base_resources(
```

Comment:
Could you elaborate? As far as I can see this method only leads to a single query. Or do you mean we could more easily make a copy with the new version? If so I think that might make the actual merge a bit more difficult because you would need to delete all resources for the sets that are present in this export.

Reply:
I would make this into another ticket as discussed

Reply:
#4743